### PR TITLE
quick entity selection using right MB click

### PR DESCRIPTION
**Pull request by ymll58**
Solves #215

### CODE REVIEW

File: modules/core/web/src/components/ModelingTool.vue [hunk 1]
```diff
@@ -1,31 +1,41 @@
...
 import Connection from "../model/diagram/Connection";
 import Line from "../model/diagram/Line";
+import { defineProps } from "vue";
+import { storeToRefs } from "pinia";
+import IconEntity from "@/components/icons/IconEntitytyp.vue";
+import IconRelationshiptyp from "@/components/icons/IconRelationshiptyp.vue";
+import IconEntityRelationshiptyp from "@/components/icons/IconEntityRelationshiptyp.vue";
+import EntityType from "@/enums/EntityType";
+const isContextMenuVisible = ref(false);
+const contextMenuPosition = ref({ x: '0px', y: '0px' });
```
Comment (Medienkunzt) [inline modules/core/web/src/components/ModelingTool.vue:38]: Variables should never be defined under the imports. Please place them between the props and onMounted().

File: modules/core/web/src/components/ModelingTool.vue [hunk 2]
```diff
@@ -265,7 +293,33 @@ const calculateGradients = (groupedEntitys: Connection[][], calculatedLines: Lin
...
+});
+const handleContextMenuItem = (item: string) => {
+  if (item === "dog") {
+    // Handle the "Dog" selection
+  } else if (item === "cat") {
+    // Handle the "Cat" selection
+  } else if (item === "mouse") {
+    // Handle the "Mouse" selection
+  }
+  isContextMenuVisible.value = false;
```
Comment (Medienkunzt) [inline modules/core/web/src/components/ModelingTool.vue:322]: pls remove unused functions

File: modules/core/web/src/components/ModelingTool.vue [hunk 3]
```diff
@@ -304,4 +358,31 @@ const calculateGradients = (groupedEntitys: Connection[][], calculatedLines: Lin
...
+
+.context-menu-item {
+  padding: 4px 0;
+  transition: background-color 0.2s;
+  cursor: pointer;
+}
+
+.context-menu-item:hover {
+  background-color: #f0f0f0;
+}
```
Comment (Medienkunzt) [inline modules/core/web/src/components/ModelingTool.vue:387]: if the vue components are used it could be possible that this is no longer needed.

**Review by Medienkunzt — CHANGES_REQUESTED** [review]
Referring to ticket #215 , the context menu was not built in the given structure (use vuetify components). This structure is important, because further issues will be based upon.

When inserting the elements, the new entity is not inserted in the correct position. The new element should be inserted exactly at the position of the mouse click.

**Review by Medienkunzt — CHANGES_REQUESTED** [review]
If you want to edit an entity with a right click, for example to add attributes, then the field for the quick insertion of elements also appears.
image

**Review by Medienkunzt — CHANGES_REQUESTED** [review]
The context menu is still not in the given structure. See figure in ticket...
image

**Comment by ymll58** [comment]
All the above mentionned requests were implemented . 
the quick insertion menu will not appear as long as the user clicks accurately  on the frame of the entities. 
double clicks on the frame will still activate the quick menu . 
image

**Review by Medienkunzt — CHANGES_REQUESTED** [review]
I've made several adjustments to ensure that all functionalities are represented as described in the ticket.

However, there's one more thing that needs to be addressed. As seen throughout the project, we're working in a component-based manner. To clearly represent the "quick insert" functionality, it should be refactored into its own component, just like all other functions. Please take into account the communication capabilities provided by Vue3 between components when doing this.

**Review by Medienkunzt — APPROVED** [review]

**Pull request merged**